nima/webclient/tracing/src/main/java/io/helidon/nima/webclient/tracing/WebClientTracing.java

```diff
@@ -0,0 +1,169 @@
+import io.helidon.tracing.Span;
+import io.helidon.tracing.SpanContext;
+import io.helidon.tracing.Tag;
+import io.helidon.tracing.Tracer;
+
+/**
+ * Client service for tracing propagation.
+ * This service adds appropriate headers to the outbound request to propagate trace across services, so the spans
+ * can form a hierarchy in a tracer, such as Jaeger.
+ */
+public class WebClientTracing implements WebClientService {
+    private final Function<Context, Tracer> tracerFunction;
+
+    private WebClientTracing() {
+        tracerFunction = ctx -> ctx.get(Tracer.class)
+                .orElseGet(Tracer::global);
+    }
+
+    private WebClientTracing(Tracer tracer) {
+        this.tracerFunction = ctx -> tracer;
+    }
+
+    /**
+     * Creates new instance of client tracing service.
+     *
+     * @return client tracing service
+     */
+    public static WebClientService create() {
+        return new WebClientTracing();
+    }
+
+    /**
+     * Creates a new client tracing service for a specific tracer.
+     *
+     * @param tracer to create new spans
+     * @return client tracing service
+     */
+    public static WebClientService create(Tracer tracer) {
+        return new WebClientTracing(tracer);
+    }
+
+    @Override
+    public WebClientServiceResponse handle(Chain chain, WebClientServiceRequest request) {
+        String method = request.method().text();
+
+        Tracer tracer = tracerFunction.apply(request.context());
+
+        Span.Builder spanBuilder = tracer.spanBuilder(composeName(method, request));
+
+        request.context().get(SpanContext.class).ifPresent(spanBuilder::parent);
+
+        spanBuilder.kind(Span.Kind.CLIENT);
+        spanBuilder.tag(Tag.COMPONENT.create("helidon-webclient"));
+        spanBuilder.tag(Tag.HTTP_METHOD.create(method));
+        spanBuilder.tag(Tag.HTTP_URL.create(request.uri().toString()));
+        Span span = spanBuilder.start();
+
+        request.context().register(span.context());
+
+        tracer.inject(span.context(),
+                      HeaderProvider.empty(),
+                      new ClientHeaderConsumer(request.headers()));
+
+        try {
+            WebClientServiceResponse response = chain.proceed(request);
+            Http.Status status = response.status();
+            span.tag(Tag.HTTP_STATUS.create(status.code()));
+
+            Http.Status.Family family = status.family();
+            if (family == Http.Status.Family.CLIENT_ERROR || family == Http.Status.Family.SERVER_ERROR) {
+                span.addEvent("error", Map.of("message",
+                                              "Response HTTP status: " + status,
+                                              "error.kind",
+                                              (family == Http.Status.Family.CLIENT_ERROR)
+                                                      ? "ClientError"
+                                                      : "ServerError"));
+            }
```

Comment:
Maybe could be more readable in standalone branches
```suggestion
            if (family == Http.Status.Family.CLIENT_ERROR) {
                span.addEvent("error", Map.of("message", "Response HTTP status: " + status, "error.kind", "ClientError"));
            }
            if (family == Http.Status.Family.SERVER_ERROR) {
                span.addEvent("error", Map.of("message", "Response HTTP status: " + status, "error.kind", "ServerError"));
            }
```

Reply:
This duplicates most of the code, where only one string differs. Thinking how to refactor